print_not_supported errors out with TypeError

[this line](https://github.com/sympy/sympy/blob/master/sympy/printing/numpy.py#L264C24-L264C44l)
(which is the return statement shown below)
cannot ever work. 
You get

```


    def _print_NDimArray(self, expr):
        if len(expr.shape) == 1:
            return self._module + '.array(' + self._print(expr.args[0]) + ')'
        if len(expr.shape) == 2:
            return self._print(expr.tomatrix())
        # Should be possible to extend to more dimensions
>       return super()._print_not_supported(self, expr)
E       TypeError: CodePrinter._print_not_supported() takes 2 positional arguments but 3 were given


```

fix should be trivial

--- a/sympy/printing/numpy.py
+++ b/sympy/printing/numpy.py
@@ -240,11 +240,11 @@ class NumPyPrinter(ArrayPrinter, PythonCodePrinter):
 
     def _print_MatrixBase(self, expr):
         if 0 in expr.shape:
-            func = self._module_format(self._module + '.zeros')
+            func = self._module_format(f'{self._module}.{self._zeros}')
             return f"{func}({self._print(expr.shape)})"
         func = self.known_functions.get(expr.__class__.__name__, None)
         if func is None:
-            func = self._module_format(self._module + '.array')
+            func = self._module_format(f'{self._module}.array')
         return "%s(%s)" % (func, self._print(expr.tolist()))
 
     def _print_Identity(self, expr):
@@ -260,12 +260,12 @@ class NumPyPrinter(ArrayPrinter, PythonCodePrinter):
 
     def _print_NDimArray(self, expr):
         if expr.rank() == 0:
-            func = self._module_format(self._module + '.array')
+            func = self._module_format(f'{self._module}.array')
             return f"{func}({self._print(expr[()])})"
         if 0 in expr.shape:
-            func = self._module_format(self._module + '.zeros')
+            func = self._module_format(f'{self._module}.{self._zeros}')
             return f"{func}({self._print(expr.shape)})"
-        func = self._module_format(self._module + '.array')
+        func = self._module_format(f'{self._module}.array')
         return f"{func}({self._print(expr.tolist())})"
 
     _add = "add"

--- a/sympy/printing/numpy.py
+++ b/sympy/printing/numpy.py
@@ -239,6 +239,9 @@ class NumPyPrinter(ArrayPrinter, PythonCodePrinter):
         return "%s(%s)" % (self._module_format(self._module + '.sinc'), self._print(expr.args[0]/S.Pi))
 
     def _print_MatrixBase(self, expr):
+        if 0 in expr.shape:
+            func = self._module_format(self._module + '.zeros')
+            return f"{func}({self._print(expr.shape)})"
         func = self.known_functions.get(expr.__class__.__name__, None)
         if func is None:
             func = self._module_format(self._module + '.array')
@@ -256,12 +259,14 @@ class NumPyPrinter(ArrayPrinter, PythonCodePrinter):
                                  self._print(expr.args[0].tolist()))
 
     def _print_NDimArray(self, expr):
-        if len(expr.shape) == 1:
-            return self._module + '.array(' + self._print(expr.args[0]) + ')'
-        if len(expr.shape) == 2:
-            return self._print(expr.tomatrix())
-        # Should be possible to extend to more dimensions
-        return super()._print_not_supported(self, expr)
+        if expr.rank() == 0:
+            func = self._module_format(self._module + '.array')
+            return f"{func}({self._print(expr[()])})"
+        if 0 in expr.shape:
+            func = self._module_format(self._module + '.zeros')
+            return f"{func}({self._print(expr.shape)})"
+        func = self._module_format(self._module + '.array')
+        return f"{func}({self._print(expr.tolist())})"
 
     _add = "add"
     _einsum = "einsum"

--- a/sympy/printing/tests/test_jax.py
+++ b/sympy/printing/tests/test_jax.py
@@ -334,7 +334,7 @@ def test_issue_17006():
 
 def test_jax_array():
     assert JaxPrinter().doprint(Array(((1, 2), (3, 5)))) == 'jax.numpy.array([[1, 2], [3, 5]])'
-    assert JaxPrinter().doprint(Array((1, 2))) == 'jax.numpy.array((1, 2))'
+    assert JaxPrinter().doprint(Array((1, 2))) == 'jax.numpy.array([1, 2])'
 
 
 def test_jax_known_funcs_consts():

--- a/sympy/printing/tests/test_numpy.py
+++ b/sympy/printing/tests/test_numpy.py
@@ -6,6 +6,7 @@ from sympy.functions.elementary.miscellaneous import sqrt
 from sympy.functions.elementary.piecewise import Piecewise
 from sympy.functions.special.gamma_functions import polygamma
 from sympy.functions.special.error_functions import (Si, Ci)
+from sympy.matrices import Matrix
 from sympy.matrices.expressions.blockmatrix import BlockMatrix
 from sympy.matrices.expressions.matexpr import MatrixSymbol
 from sympy.matrices.expressions.special import Identity
@@ -329,8 +330,23 @@ def test_jax_tuple_compatibility():
     assert np.allclose(func(*input_tuple2), func(*input_array2))
 
 def test_numpy_array():
-    assert NumPyPrinter().doprint(Array(((1, 2), (3, 5)))) == 'numpy.array([[1, 2], [3, 5]])'
-    assert NumPyPrinter().doprint(Array((1, 2))) == 'numpy.array((1, 2))'
+    p = NumPyPrinter()
+    assert p.doprint(Array([[1, 2], [3, 5]])) == 'numpy.array([[1, 2], [3, 5]])'
+    assert p.doprint(Array([1, 2])) == 'numpy.array([1, 2])'
+    assert p.doprint(Array([[[1, 2, 3]]])) == 'numpy.array([[[1, 2, 3]]])'
+    assert p.doprint(Array([], (0,))) == 'numpy.zeros((0,))'
+    assert p.doprint(Array([], (0, 0))) == 'numpy.zeros((0, 0))'
+    assert p.doprint(Array([], (0, 1))) == 'numpy.zeros((0, 1))'
+    assert p.doprint(Array([], (1, 0))) == 'numpy.zeros((1, 0))'
+    assert p.doprint(Array([1], ())) == 'numpy.array(1)'
+
+def test_numpy_matrix():
+    p = NumPyPrinter()
+    assert p.doprint(Matrix([[1, 2], [3, 5]])) == 'numpy.array([[1, 2], [3, 5]])'
+    assert p.doprint(Matrix([1, 2])) == 'numpy.array([[1], [2]])'
+    assert p.doprint(Matrix(0, 0, [])) == 'numpy.zeros((0, 0))'
+    assert p.doprint(Matrix(0, 1, [])) == 'numpy.zeros((0, 1))'
+    assert p.doprint(Matrix(1, 0, [])) == 'numpy.zeros((1, 0))'
 
 def test_numpy_known_funcs_consts():
     assert _numpy_known_constants['NaN'] == 'numpy.nan'

--- a/sympy/printing/tests/test_str.py
+++ b/sympy/printing/tests/test_str.py
@@ -1168,6 +1168,10 @@ def test_NDimArray():
     assert sstr(NDimArray(1.0), full_prec=False) == '1.0'
     assert sstr(NDimArray([1.0, 2.0]), full_prec=True) == '[1.00000000000000, 2.00000000000000]'
     assert sstr(NDimArray([1.0, 2.0]), full_prec=False) == '[1.0, 2.0]'
+    assert sstr(NDimArray([], (0,))) == 'ImmutableDenseNDimArray([], (0,))'
+    assert sstr(NDimArray([], (0, 0))) == 'ImmutableDenseNDimArray([], (0, 0))'
+    assert sstr(NDimArray([], (0, 1))) == 'ImmutableDenseNDimArray([], (0, 1))'
+    assert sstr(NDimArray([], (1, 0))) == 'ImmutableDenseNDimArray([], (1, 0))'
 
 def test_Predicate():
     assert sstr(Q.even) == 'Q.even'

--- a/sympy/tensor/array/ndim_array.py
+++ b/sympy/tensor/array/ndim_array.py
@@ -362,7 +362,8 @@ class NDimArray(Printable):
 
         if self.rank() == 0:
             return printer._print(self[()])
-
+        if 0 in self.shape:
+            return f"{self.__class__.__name__}([], {self.shape})"
         return f(self._loop_size, self.shape, 0, self._loop_size)
 
     def tolist(self):

